From d2497a263df4e1dcf209e3fc6fea2c8ff6fc0f14 Mon Sep 17 00:00:00 2001
From: Jianxu Zheng <jian.xu.zheng@intel.com>
Date: Thu, 29 Jan 2015 15:43:12 +0800
Subject: [PATCH 240/255] ov5693: stop streaming after set fmt

ov5693 may have stream out of it after set basic setting to its
registers. This would cause ISP timeout because it's not ready
to receive data yet.

This patch add another stream off call after set the settings.

Change-Id: Ic6814d92cc7286bba1bca897fd525aa8ee792078
Signed-off-by: Jianxu Zheng <jian.xu.zheng@intel.com>
Reviewed-on: https://icggerrit.ir.intel.com/38256
Reviewed-by: Jing Tao <jing.tao@intel.com>
Approver: Wei Tao <wei.tao@intel.com>
Reviewed-by: Wei Tao <wei.tao@intel.com>
---
 .../camera/drivers/media/i2c/ov5693-ecs/ov5693.c               | 10 ++++++++++
 1 file changed, 10 insertions(+)

diff --git a/drivers/external_drivers/camera/drivers/media/i2c/ov5693-ecs/ov5693.c b/drivers/external_drivers/camera/drivers/media/i2c/ov5693-ecs/ov5693.c
index 45c7917..c0b4c36 100644
--- a/drivers/external_drivers/camera/drivers/media/i2c/ov5693-ecs/ov5693.c
+++ b/drivers/external_drivers/camera/drivers/media/i2c/ov5693-ecs/ov5693.c
@@ -1634,6 +1634,16 @@ static int ov5693_s_mbus_fmt(struct v4l2_subdev *sd,
 	if (ret)
 		dev_err(&client->dev, "ov5693 startup err\n");
 
+	/*
+	 * After sensor settings are set to HW, sometimes stream is started.
+	 * This would cause ISP timeout because ISP is not ready to receive
+	 * data yet. So add stop streaming here.
+	 */
+	ret = ov5693_write_reg(client, OV5693_8BIT, OV5693_SW_STREAM,
+				OV5693_STOP_STREAMING);
+	if (ret)
+		dev_warn(&client->dev, "ov5693 stream off err\n");
+
 	ret = ov5693_get_intg_factor(client, ov5693_info,
 					&ov5693_res[dev->fmt_idx]);
 	if (ret) {
-- 
1.9.1

